src/YPerl.cc

```diff
@@ -900,7 +900,8 @@ YPerl::fromPerlScalar( SV * sv, constTypePtr wanted_type )
 	    // instead of "if (SvXOK (sv)) SvXV (sv)"
 	    // we first SvXV (sv) and only then SvXOK.
 	    const char *pv = SvPV_nolen (sv);
-	    if (SvPOK (sv))
+
+            if ( SvPOK(sv) || SvIOK(sv) || SvNOK(sv) ) // bsc#1200990
```

Comment:
Thanks!
I'll have to refresh my knowledge of the embedding API, but even now I can tell that the comment about coercion will need updating too.
Ah, the bug report quotes the relevant part of perldelta, we will want to include it in the commit message.